pkg/boot/multiboot/reader.go

```diff
@@ -60,3 +67,14 @@ func readFile(name string) ([]byte, error) {
 
 	return ioutil.ReadAll(f)
 }
+
+// TODO: could be tryCompressionFilters, grub support gz,xz and lzop
+// TODO: do we want to keep the filter inside multiboot? This could be the responsibility of the caller...
+func tryGzipFilter(r io.ReaderAt) io.ReaderAt {
+	b, err := readGzip(uio.Reader(r))
+	if err == nil {
+		kernel := kernelReader{buf: b}
```

Comment:
why is kernelReader a thing? could it just be bytes.Buffer?

Reply:
I kept the existing code, but you're right I should simplify it, I'll try bytes.Buffer, that sounds good (and already implement the inMemReaderAt interface) !

Reply:
Except bytes.Buffer is not a ReaderAt, bytes.Reader is one (but not a inMemReaderAt)...
I replaced the optimization with using io.WriterTo interface when possible.